test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -214,6 +265,33 @@ TEST_P(ProxyProtocolTest, PartialRead) {
   disconnect();
 }
 
+TEST_P(ProxyProtocolTest, v2PartialRead) {
+  // header[12]
+  // ver_cmd[1]
+  // fam[1]
+  // len[2]
+  // ip,port 1.2.3.4:0001
+  // ip,port 1.2.3.5:0002
+  constexpr uint8_t buffer[] = {0x0d, 0x0a, 0x0d, 0x0a, 0x00, 0x0d, 0x0a, 0x51, 0x55,
+                                0x49, 0x54, 0x0a, 0x21, 0x11, 0x00, 0x0c, 0x01, 0x02,
+                                0x03, 0x04, 0x00, 0x01, 0x01, 0x02, 0x03, 0x05, 0x00,
+                                0x02, 'm',  'o',  'r',  'e',  'd',  'a',  't',  'a'};
+  connect();
+
+  for (size_t i = 0; i < sizeof(buffer); i += 9) {
+    write(&buffer[i], 9);
+    if (i == 0)
+      dispatcher_.run(Event::Dispatcher::RunType::NonBlock);
+  }
+
+  expectData("moredata");
+
+  EXPECT_EQ(server_connection_->remoteAddress()->ip()->addressAsString(), "1.2.3.4");
+  EXPECT_TRUE(server_connection_->localAddressRestored());
+
+  disconnect();
+}
+
```

Comment:
can we add some additional tests to cover the error handling? [coverage report](https://67004-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.source_extensions_filters_listener_proxy_protocol_proxy_protocol.cc.html)